src/main/java/jenkins/branch/Branch.java

```diff
@@ -42,6 +42,7 @@
 
 /**
  * A source code branch.
+ * Not to be subclassed outside this plugin.
```

Comment:
Never mind, it is not possible, since the regular constructor is callable as `public` so we cannot create a `protected` constructor and restrict it.